src/engine/enginebuffer.h

```diff
@@ -346,11 +347,12 @@ class EngineBuffer : public EngineObject {
 
     // Object used to perform waveform scaling (sample rate conversion)
     EngineBufferScale* m_pScale;
+    FRIEND_TEST(EngineBufferTest, SlowRubberBand);
     // Object used for linear interpolation scaling of the audio
-    EngineBufferScaleLinear* m_pScaleLinear;
+    EngineBufferScale* m_pScaleLinear;
     // Object used for pitch-indep time stretch (key lock) scaling of the audio
-    EngineBufferScaleST* m_pScaleST;
-    EngineBufferScaleRubberBand* m_pScaleRB;
+    EngineBufferScale* m_pScaleST;
+    EngineBufferScale* m_pScaleRB;
```

Comment:
It's necessary so that the scalers can be mocked in tests.  I need to be able to replace the scalers so I can see that they are being manipulated correctly, and since there is only one place in enginebuffer that relies on child class behavior, it seems fine to have all the pointers be base-class pointers.